libsoftwarecontainer/src/gateway/networkgateway.cpp

```diff
@@ -41,28 +40,173 @@ NetworkGateway::~NetworkGateway()
 
 ReturnCode NetworkGateway::readConfigElement(const json_t *element)
 {
-    if (!read(element, "internet-access", m_internetAccess)) {
-        log_error() << "either key \"internet-access\" missing or not a bool in json configuration";
+    Entry e;
+    if (!read(element, "type", e.type)) {
+        log_error() << "No type specified in network config.";
         return ReturnCode::FAILURE;
     }
 
-    if (m_internetAccess) {
-        log_info() << "Internet access will be enabled";
+    if (e.type != "INCOMING" && e.type != "OUTGOING") {
+        log_error() << e.type << " is not a valid type ('INCOMING' or 'OUTGOING')";
+        return ReturnCode::FAILURE;
+    }
+
+    const json_t *prio = json_object_get(element, "priority");
```

Comment:
What do you mean? Isn't the JSONParser a wrapper around jansson? Which we use elsewhere when parsing json?